### PR TITLE
Upgrade to OE 0.19.0-rc1

### PR DESCRIPTION
**Pull request by eddyashton**
TODO:
- [x] Build new CI image (https://github.com/microsoft/CCF/actions/runs/4678101373)
- [x] Update references to use new CI image 
- [x] Check test results
- [x] Describe upgrade in CHANGELOG

### CODE REVIEW

File: getting_started/setup_vm/roles/ccf_build/vars/clang11.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  - apt-transport-https
+  - ninja-build
+  - libuv1-dev
+  - libc++-{{ clang_ver }}-dev
+  - libc++abi-{{ clang_ver }}-dev
+  - python3.8-dev
+  - python3.8-venv
+  - llvm-{{ clang_ver }}
+  - clang-{{ clang_ver }}
+  - clang-format-10 # On purpose, to avoid formatting conflicts
```
Comment (achamayou) [inline getting_started/setup_vm/roles/ccf_build/vars/clang11.yml:14]: Ugh. We'll probably have to bite the bullet and do a reformat with 11, we don't want to stay stuck on format-10 forever.

File: getting_started/setup_vm/roles/openenclave/vars/common.yml [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
 oe_deb: "https://github.com/openenclave/openenclave/releases/download/v{{ oe_ver }}/Ubuntu_2004_open-enclave_{{ oe_ver_ }}_amd64.deb"
 oe_host_verify_deb: "https://github.com/openenclave/openenclave/releases/download/v{{ oe_ver }}/Ubuntu_2004_open-enclave-hostverify_{{ oe_ver_ }}_amd64.deb"
 
 # LVI mitigations
 oe_lvi_scripts_dir: "{{ oe_prefix }}/bin/scripts/lvi-mitigation"
-oe_lvi_clang_version: "clang-10"
-oe_lvi_clangpp_version: "clang++-10"
+oe_lvi_clang_version: "clang-11"
+oe_lvi_clangpp_version: "clang++-11"
 oe_lvi_bin_dir: /opt/oe_lvi
```
Comment (achamayou) [inline getting_started/setup_vm/roles/openenclave/vars/common.yml:21]: We shouldn't need those anymore, same with the scripts dir.

File: cmake/open_enclave.cmake [hunk 3]
```diff
@@ -26,30 +26,28 @@ if(COMPILE_TARGET STREQUAL "sgx")
 
   option(LVI_MITIGATIONS "Enable LVI mitigations" ON)
   if(LVI_MITIGATIONS)
+    set(LVI_MITIGATION_BINDIR
```
Comment (achamayou) [inline cmake/open_enclave.cmake:29]: That should be unnecessary now, clang 11+ supports the flag directly.

File: getting_started/setup_vm/roles/ccf_build/vars/clang11.yml (hunk 1 above)
Comment (jumaffre) [reply getting_started/setup_vm/roles/ccf_build/vars/clang11.yml:14]: #5181

**Comment by achamayou** [comment]
@jumaffre we should also update the copy of snmalloc we use on the virtual side.

File: .azure-pipelines-templates/ninja.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
 steps:
-  - script: ninja
+  - script: ninja --verbose
```
Comment (jumaffre) [inline .azure-pipelines-templates/ninja.yml:2]:
```suggestion
  - script: ninja
```

**Review by achamayou — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 [4.0.0-rc2]: https://github.com/microsoft/CCF/releases/tag/ccf-4.0.0-rc2
 
+### Changed
+
+- Updated Open Enclave to 0.19.0 (#5165).
```
Comment (achamayou) [inline CHANGELOG.md:14]: @jumaffre can we mention the clang upgrade for SGX builds.

**Pull request merged**